app/form_builders/adp_text_field.rb

```diff
@@ -0,0 +1,134 @@
+
+# class to create specialized text field wrapped in all the GDS gubbins
+class AdpTextField
+
+  include ExternalUsers::ClaimsHelper
+  include ActionView::Helpers::TagHelper
+
+
+  # instantiate an AdpTextField object
+  # * form: the instance of AdpFormBuilder that this is called from
+  # * method: the method on the object that the form is wrapping that this input field is for
+  # * options:
+  #   * label: label to be provided for the input field
+  #   * hint_text: Hint text displayed underneath the label
+  #   * errors: An ErrorPresenter for the form object, or the form object itself
+
+  def initialize(form, method, options)
+    @form = form
+    @method = method
+    @options = options
+    @form_field_id = generate_form_field_id
+    @form_field_name = generate_form_field_name
+    @errors = options[:errors]
+    @anchor_id = generate_anchor_id
+  end
+
+
+  # the methods output_buffer= and output_buffer are required by the content_tag
+  # methods, called from the validation_error_message in ExternalUsers::ClaimsHelper
+  #
+  def output_buffer=(value)
+    @errors = value
+  end
+
+  def output_buffer
+    @errors
+  end
+
+  def has_errors?
+    @errors.errors_for?(@anchor_id.to_sym)
+  end
+
+  def to_html
+    result = div_start
+    result += anchor
+    result += label
+    result += hint
+    result += error_message
```

Comment:
The error message and the span it is wrapped in should only be shown if there is an error.

Currently this element is being shown and makes for the red line

```html
<span class="error"></span>
```